main_test.go

```diff
@@ -531,27 +556,38 @@ func TestEventRPCNotifier(t *testing.T) {
 	shardID := ShardIDType(1)
 	period := 2
 
-	// NotifyCollation
+	// send collation to the Host side
 	collation := &pbmsg.Collation{
 		ShardID: shardID,
 		Period:  PBInt(period),
 		Blobs:   []byte("123"),
 	}
-	isValid, err := eventNotifier.NotifyCollation(collation)
+	collationBytes := protoToBytes(t, collation)
+	isValidBytes, err := eventNotifier.Receive("", typeCollation, collationBytes)
 	if err != nil {
 		if err != nil {
-			t.Errorf("something wrong when calling `eventNotifier.NotifyCollation`: %v", err)
+			t.Errorf("something wrong when calling `eventNotifier.Receive`: %v", err)
 		}
 	}
-	if !isValid {
-		t.Errorf("wrong reponse from `eventNotifier.NotifyCollation`")
+	if len(isValidBytes) != 1 && isValidBytes[0] != 0 {
+		t.Errorf("wrong reponse from `eventNotifier`")
 	}
 
 	// GetCollation
-	collationHash := "123"
-	collationReceived, err := eventNotifier.GetCollation(shardID, period, collationHash)
+	req := &pbmsg.CollationRequest{
+		ShardID: shardID,
+		Period:  PBInt(period),
+		Hash:    "123",
+	}
+	reqBytes := protoToBytes(t, req)
+	respBytes, err := eventNotifier.Receive("", typeCollationRequest, reqBytes)
+	if err != nil {
+		t.Errorf("something wrong when calling `eventNotifier.Receive`: %v", err)
+	}
+	collationReceived := &pbmsg.Collation{}
+	err = proto.Unmarshal(respBytes, collationReceived)
 	if err != nil {
-		t.Errorf("something wrong when calling `eventNotifier.GetCollation`: %v", err)
+		t.Errorf("failed to unmarshal collation bytes")
```

Comment:
seems it worth extracting these few lines to a function `requestCollation` for test, to avoid DRY

Reply:
IMO it seems not necessary for now, and can be refactored further later.